### PR TITLE
feat: parse Cuda compute cap from env

### PR DESCRIPTION
**Pull request by OlivierDehaene**
Parse compute cap from env and fallback to nvidia-smi if not found instead of the other way around.
This is needed for docker builds.

### CODE REVIEW

**Comment by LaurentMazare** [comment]
If it's easy to do on your side, would you mind splitting this PR in the actual change vs the reformatting? I feel that this would make the diff a lot easier to read.

File: candle-flash-attn/build.rs [hunk 1]
```diff
@@ -152,18 +166,6 @@ fn main() -> Result<()> {
     println!("cargo:rustc-link-lib=dylib=cudart");
     println!("cargo:rustc-link-lib=dylib=stdc++");
 
-    /* laurent: I tried using the cc cuda integration as below but this lead to ptaxs never
```
Comment (LaurentMazare) [inline candle-flash-attn/build.rs:155]: Could we keep this comment, it's probably useful to know that this was tried if someone wants to pick it up again.

File: candle-kernels/build.rs [hunk 2]
```diff
@@ -141,83 +222,22 @@ mod cuda {
                 }
             }
             codes.sort();
-            if !codes.contains(&compute_cap) {
-                panic!("nvcc cannot target gpu arch {compute_cap}. Available nvcc targets are {codes:?}.");
-            }
-            *codes.last().unwrap()
+            let max_nvcc_code = *codes.last().unwrap();
```
Comment (LaurentMazare) [inline candle-kernels/build.rs:225]: Same as above, an explicit error might be better.

**Comment by OlivierDehaene** [comment]
Thanks for the review! I pushed some modifications.

**Comment by LaurentMazare** [comment]
Thanks!

**Pull request merged**